p2p/cryptoBox/box.go

```diff
@@ -0,0 +1,159 @@
+package cryptoBox
+
+import (
+	crypto_rand "crypto/rand"
+	"errors"
+	"fmt"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"golang.org/x/crypto/nacl/box"
+	"io"
+)
+
+const (
+	keySize = 32 // non-configurable, expected by NaCl
+	nonceSize = 24 // non-configurable, expected by NaCl
+)
+
+type Key interface {
+	Bytes() []byte
+	Raw() *[keySize]byte
+	String() string
+	Pretty() string
+}
+
+type PrivateKey interface {
+	Key
+}
+
+type PublicKey interface {
+	Key
+}
+
+type SharedSecret interface {
+	Key
+	Seal(message []byte) (out []byte)
+	Open(encryptedMessage []byte) (out []byte, err error)
+}
+
+type key struct {
+	bytes *[keySize]byte
+}
+
+var _ PrivateKey = &key{}
```

Comment:
its better to replace these with `(*key)(nil)` since using `&key` probably allocates memory.

Reply:
done